src/core/Akka/Actor/ActorRef.cs

```diff
@@ -140,7 +166,35 @@ protected override void TellInternal(object message, IActorRef sender)
             if (!handled && !_result.Task.IsCanceled)
                 _provider.DeadLetters.Tell(message ?? default(T), this);            
         }
-        
+
+        public override void SendSystemMessage(ISystemMessage message)
+        {
+            if (message is Watch watch)
+            {
+                if (_result.Task.IsCompleted)
```

Comment:
Automatically covers any possible completion state for the task: cancelled, faulted, or ran to completion